### PR TITLE
Fixed #33084 -- Removed incorrect system check for ManyToManyField with limit_choices_to.

### PR DESCRIPTION
**Pull request by hramezani**
https://code.djangoproject.com/ticket/33084

### CODE REVIEW

File: tests/invalid_models_tests/test_relative_fields.py [hunk 1]
```diff
@@ -120,12 +120,6 @@ class ThroughModel(models.Model):
                 obj=field,
                 id='fields.W341',
             ),
-            DjangoWarning(
```
Comment (hramezani) [inline tests/invalid_models_tests/test_relative_fields.py:123]: Do we need to remove `ThroughModel` and `through='ThroughModel'` in `ModelM2M.m2m`?

**Comment by charettes** [comment]
We should adjust [the documentation](https://github.com/django/django/blob/1bd6a7a0acc11e249fca11c017505ad39f15ebf6/docs/ref/models/fields.txt#L1813-L1815) and likely add a regression test to confirm the behaviour works.

**Comment by hramezani** [comment]
Thanks, @charettes for the review.

> We should adjust [the documentation](https://github.com/django/django/blob/1bd6a7a0acc11e249fca11c017505ad39f15ebf6/docs/ref/models/fields.txt#L1813-L1815) 

I've removed this part of the doc based on the ticket. which adjustment do you mean?

> and likely add a regression test to confirm the behaviour works.

 I've added @felixxm test from the ticket to the patch.

**Comment by charettes** [comment]
> which adjustment do you mean?

I meant the removal, thanks for that!

File: docs/ref/checks.txt [hunk 2]
```diff
@@ -309,8 +309,6 @@ Related fields
 * **fields.W341**: ``ManyToManyField`` does not support ``validators``.
 * **fields.W342**: Setting ``unique=True`` on a ``ForeignKey`` has the same
   effect as using a ``OneToOneField``.
-* **fields.W343**: ``limit_choices_to`` has no effect on ``ManyToManyField``
-  with a ``through`` model.
```
Comment (felixxm) [inline docs/ref/checks.txt:313]: We don't want to reuse the same number in the future, so the docs should stay. Please add a note:
```
* **fields.W343**: ``limit_choices_to`` has no effect on ``ManyToManyField``
  with a ``through`` model. *This check appeared before Django 4.0.*
```

File: tests/invalid_models_tests/test_relative_fields.py (hunk 1 above)
Comment (felixxm) [reply tests/invalid_models_tests/test_relative_fields.py:123]: Yes, I'd remove them from this test method and add a separate test, e.g.
```python
    def test_many_to_many_with_limit_choices_through_no_warning(self):
        class Model(models.Model):
            name = models.CharField(max_length=20)

        class ModelM2M(models.Model):
            m2m = models.ManyToManyField(
                Model,
                limit_choices_to={'name': 'test_name'},
                through='ThroughModel',
                through_fields=('modelm2m', 'model'),
            )

        class ThroughModel(models.Model):
            model = models.ForeignKey('Model', models.CASCADE)
            modelm2m = models.ForeignKey('ModelM2M', models.CASCADE)

        self.assertEqual(ModelM2M.check(), [])
```

File: tests/m2m_through/tests.py [hunk 3]
```diff
@@ -388,6 +389,21 @@ def test_through_fields(self):
             attrgetter('name')
         )
 
+    def test_limit_choices_through(self):
```
Comment (felixxm) [inline tests/m2m_through/tests.py:392]: Please move this test to the `model_forms.tests.LimitChoicesToTests`.

File: tests/m2m_through/tests.py [hunk 4]
```diff
@@ -388,6 +389,21 @@ def test_through_fields(self):
...
+        for i in range(-10, 10):
+            Number.objects.create(value=i)
```
Comment (felixxm) [inline tests/m2m_through/tests.py:394]: It should be enough to add 2-3 numbers, e.g. `0`, `1`, `2`.

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks :+1: I left comments.

**Review by felixxm — COMMENTED** [review]
@hramezani Thanks :+1:

**Pull request merged**